src/plugins/dashboard/public/url_generator.ts

```diff
@@ -99,7 +132,7 @@ export const createDirectAccessDashboardLinkGenerator = (
       GLOBAL_STATE_STORAGE_KEY,
       cleanEmptyKeys({
         time: state.timeRange,
-        filters: state.filters?.filter(f => esFilters.isFilterPinned(f)),
+        filters: filters?.filter(f => esFilters.isFilterPinned(f)),
```

Comment:
I know this is not a new change, but for my own understanding - why do we need to filter pinned filters here?

Reply:
`url_generator` accepts array of filters to apply and to make `pinned` state work in the url properly we have to split those into `_a` and `_g` :(